eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1540,4 +1540,57 @@ public function removeFieldDefinitionTranslation(
 
         $deleteQuery->execute();
     }
+
+    /**
+     * Removes types created or modified by the user.
+     */
+    public function removeByUserAndVersion(int $userId, int $version): void
+    {
+        $this->connection->beginTransaction();
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb->delete('ezcontentclass')
+            ->where('creator_id = :user or modifier_id = :user')
+            ->andWhere('version = :version')
+            ->setParameter('user', $userId, ParameterType::INTEGER)
+            ->setParameter('version', $version, ParameterType::INTEGER);
+
+        $qb->execute();
+
+        $this->cleanupAssociations();
+
+        $this->connection->commit();
```

Comment:
If you use beginTransaction/commit you should also use rollback. Please wrap this by try-catch, catch exception that might come from `execute` and do a rollback inside catch, then rethrow.